service/src/main/java/io/camunda/service/cache/ProcessElementProvider.java

```diff
@@ -30,27 +30,41 @@ public ProcessElementProvider(final ProcessDefinitionServices processDefinitionS
   public void extractElementNames(
       final Long processDefinitionKey,
       final BiConsumer<Long, ProcessElement> processDefinitionKeyElementConsumer) {
-    final var processDefinition = processDefinitionServices.getByKey(processDefinitionKey);
-    extractElementNames(processDefinition, processDefinitionKeyElementConsumer);
+    try {
+      final var processDefinition = processDefinitionServices.getByKey(processDefinitionKey);
+      extractElementNames(processDefinition, processDefinitionKeyElementConsumer);
+    } catch (final RuntimeException e) {
+      // TODO check what exceptions can be thrown here
```

Comment:
this is something we should discuss 😅 what should happen when the `processDefinitionServices` is unable to find the definition or fails due to network glitches or the like. I couldn't find a clear exception strategy, it was probably fully failing before? 🤔

Reply:
Interesting. Depends a bit on how `load` is called and handled when it runs into an exception in a Caffeine cache. But yeah, logging here that we couldn't fetch should actually be enough. We can't do anything about it, really. We cannot load the name into the cache, period. We could retry, if there is anything like that in a Caffeine cache (but I am not sure this exists).

@danielkelemen, any ideas, since you initially looked into this? Maybe you still remember.

Reply:
My thoughts was also more like, should the full user request fail or should we return an entity without name 🤔

Reply:
Good questions. I'd probably go for an empty name here.